### PR TITLE
rgw: handle error return value in read_form_part_header

### PR DESCRIPTION
**Pull request by wangzhengyong**
rgw: handle error return value in read_form_part_header

### CODE REVIEW

**Comment by wangzhengyong** [comment]
please retest

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -1589,6 +1589,8 @@ int RGWPostObj_ObjStore_S3::read_form_part_header(struct post_form_part *part,
       break;
 
     r = read_line(bl, chunk_size, &reached_boundary, done);
+    if (r < 0)
+        return r;
```
Comment (yehudasa) [inline src/rgw/rgw_rest_s3.cc]: @wangzhengyong looks good. It looks like we probably also need to add to check `if (*done)` like above. Maybe change the loop to `while (!*done)`?

Comment (wangzhengyong) [inline src/rgw/rgw_rest_s3.cc]: @yehudasa done

**Comment by oritwas** [comment]
Jenkins test this please

**Pull request closed without merging**